pages/docs/reference/python/guides/pydantic.mdx

```diff
@@ -0,0 +1,66 @@
+# Pydantic
+
+This guide will help you use Pydantic to perform runtime type validation when sending and receiving events.
+
+## Sending events
+
+Create a base class that all your event classes will inherit from. This class has methods to convert to and from `inngest.Event` objects.
+```py
+import inngest
+import pydantic
+import typing
+
+TEvent = typing.TypeVar("TEvent", bound="BaseEvent")
+
+class BaseEvent(pydantic.BaseModel):
+    data: pydantic.BaseModel
+    id: str = ""
+    name: typing.ClassVar[str]
+    ts: int = 0
+
+    @classmethod
+    def from_event(cls: type[TEvent], event: inngest.Event) -> TEvent:
+        return cls.model_validate(event.model_dump(mode="json"))
+
+    def to_event(self) -> inngest.Event:
+        return inngest.Event(
+            name=self.name,
+            data=self.data.model_dump(mode="json"),
+            id=self.id,
+            ts=self.ts,
+        )
+```
+
+Next, create a Pydantic model for your event.
+```py
+class MyEventData(pydantic.BaseModel):
+    count: int
+
+class MyEvent(BasePydanticEvent):
+    data: MyEventData
+    name: str = "my-event"
+```
+
+Since Pydantic validates on instantiation, the following code will raise an error if the data is invalid.
+```py
+client.send(
+    MyEvent(
+        data=MyEventData(count="bad data"),
+    ).to_event()
+)
```

Comment:
Can we use a real life event name? We've gotten feedback in the past that real event names and function names are a lot better to learn than "hello world" type stuff. What about:

```py
class PostUpvotedEventData(pydantic.BaseModel):
    count: int

class PostUpvotedEvent(BaseEvent):
    data: MyEventData
    name: str = "forum/post.upvoted"
```

Since Pydantic validates on instantiation, the following code will raise an error if the data is invalid.
```py
client.send(
    PostUpvotedEvent(
        data=PostUpvotedEventData(count="bad data"),
    ).to_event()
)
```